src/main/java/com/toptrumps/db/Statistics.java

```diff
@@ -13,12 +10,58 @@
     private int drawsAverage;
     private int maxRounds; 
 
-    public Statistics(ResultSet resultSet)throws SQLException{
-        this.gamesPlayed = resultSet.getInt(1);
-        this.aiWins = resultSet.getInt(2);
-        this.humanWins = resultSet.getInt(3);
-        this.drawsAverage = resultSet.getInt(4);
-        this.maxRounds = resultSet.getInt(5);
+    public Statistics(){
+        try{
+            buildStatistics(retrieveStats());
+        }catch(SQLException e){e.printStackTrace();}
+    }
+
+    public void persistData(GameStateCollector gameState){
+        
+        IndividualGameDAOImpl igdi = new IndividualGameDAOImpl(gameState);
+        igdi.create();
+        PerformanceDAOImpl pdi = new PerformanceDAOImpl(gameState);
+        pdi.create();
+    }
+
+    public ResultSet retrieveStats(){
+
+        Connection conn = ConnectionFactory.getConnection();
+        PreparedStatement playedStats = null;
+        ResultSet rs = null;
+        String playedStatsSQL = 
+
+        "SELECT (SELECT COUNT (game_id)" +
```

Comment:
I don't think it's necessary, but if this SQL statement can be moved to the external .sql file - it would be more readable (because in that case, all those line concatenations won't be needed)

Reply:
later

Reply:
@Arden488 I also suggested it a couple of times :p

Reply:
But we should also be careful with I/O!